### PR TITLE
BUG: datetime64 - Timestamp incorrectly raising TypeError

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/scalar/timestamp/test_arithmetic.py [hunk 1]
```diff
@@ -66,6 +66,17 @@ def test_delta_preserve_nanos(self):
...
         assert result.nanosecond == val.nanosecond
 
+    def test_rsub_dtscalars(self, tz_naive_fixture):
+        # In particular, check that datetime64 - Timestamp works GH#28286
+        td = Timedelta(1235345642000)
+        ts = Timestamp.now(tz_naive_fixture)
+        other = ts + td
+
+        assert other - ts == td
+        assert other.to_pydatetime() - ts == td
```
Comment (WillAyd) [inline pandas/tests/scalar/timestamp/test_arithmetic.py]: Is anything above this line not tested elsewhere? Just was a little counter intuitive to read as I thought this PR was only about datetime64 obects

Comment (jbrockmendel) [reply pandas/tests/scalar/timestamp/test_arithmetic.py]: trying to follow the pattern in these tests of testing all datetime-like scalars in the same place

Comment (WillAyd) [reply pandas/tests/scalar/timestamp/test_arithmetic.py]: Makes sense

File: pandas/tests/scalar/timestamp/test_arithmetic.py [hunk 2]
```diff
@@ -66,6 +66,17 @@ def test_delta_preserve_nanos(self):
...
+        if tz_naive_fixture is None:
```
Comment (WillAyd) [inline pandas/tests/scalar/timestamp/test_arithmetic.py]: What should happen if this is not None? Should there be a test for this?

**Comment by jbrockmendel** [comment]
Travis has been done for a few hours, no idea why it’s stuck on yellow

**Comment by jreback** [comment]
lgtm. can you rebase and push again.

**Comment by jbrockmendel** [comment]
rebased+green

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**